### DISCUSSION TITLE
Is it possible to encrypt all the websocket data traffic shown in the browser

### DISCUSSION THREAD

**Discussion by babashark** [nats-io/nats.ws#250] [Q&A] [answered]
I can see all the nats protocol message in the browser
<img width="756" alt="image" src="https://github.com/user-attachments/assets/47a95604-bcc9-420f-a82e-5010e2502b5d">

but I want to do something like this
<img width="819" alt="image" src="https://github.com/user-attachments/assets/dc913c41-83a7-449b-aae4-45d05f3dfee7">

**Comment by aricart** [comment, marked as answer]
It is not possible to encrypt the protocol. Secrets are not exposed in the connect or other protocols. However if your payloads have data you don't want the browser user to see, you can certainly encrypt payloads in your message.
